### PR TITLE
MetaGeneratorMiddleware #371

### PR DESCRIPTION
**Pull request by agriffard**
Add a middleware in Orchard.Commons to insert automatically a <meta content="Orchard" name="Generator"> thanks to ResourceManager.

### CODE REVIEW

**Comment by sebastienros** [comment]
Can you check the lifetime of the middleware, i.e. how many times is it instantiated, once per request?

**Comment by agriffard** [comment]
Yes, once per request, even for .css and .js files, which is bad.
How can I filter for only the needed requests ?

**Comment by sebastienros** [comment]
If the middleware is registered after the static file middlewares then it won't be called. Or right before the Route one.
I wonder what drives the lifetime of the middlewares. For instance in the tenant pipeline I know I cache them, but I don't know how it's handled for the other ones, I would have expected them to be singletons too.

File: src/Orchard.Cms.Web/Modules/Orchard.Commons/Middlewares/MetaGeneratorMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        {
+            _next = next;
+            _resourceManager = resourceManager;
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+            var metaEntry = new MetaEntry("Generator", "Orchard");
+            _resourceManager.RegisterMeta(metaEntry);
+            await _next.Invoke(httpContext);
```
Comment (Jetski5822) [inline src/Orchard.Cms.Web/Modules/Orchard.Commons/Middlewares/MetaGeneratorMiddleware.cs]: do we need async await? cant we just return the task?

**Comment by sebastienros** [comment]
As per my comment on the issue, use a filter so we are sure it's only done for a view. Still in the Commons module.

**Pull request closed without merging**